### DISCUSSION TITLE
Best way to use `make_reduction`, `WindowSummarizer` and `RandomForestRegressor`

### DISCUSSION THREAD

**Discussion by meraldoantonio** [alan-turing-institute/sktime#6739] [Q&A]
I have a macroeconomic panel data , which I'll approximate as `longley` as illustration.
Suppose I want to predict `TOTEMP` (`y`) from features derived from the lags of `TOTEMP` and all the other variables in the dataframe (`X`) using `RandomForestRegressor`
During forecasting time, I will not have contemporaneous or future values for the other variables (i.e. I won't have `X_test` during predict). I want the forecasting to proceed recursively.

My question is - how should I construct my pipeline? 

I've tried these two approaches which didn't work:

In the first, I wrapped `WindowSummarizer` and `RandomForestRegressor` within `ForecastingPipeline`. I was able to fit this to data but I couldn't perform `predict` without `X_test`

```python

regressor = RandomForestRegressor()
forecaster = make_reduction(regressor, window_length=3, strategy="recursive")

kwargs = {
    "lag_feature": {
        "lag": range(1, 3),
        "mean": [[1, 3]],
        "std": [[1, 4]],
    },
    "target_cols": ALL_CATEGORIES # all X columns
}
pipe = ForecastingPipeline(
    steps=[
        ("feature_generator", WindowSummarizer(**kwargs)),
        ("forecaster", forecaster),
    ]
)

pipe_return = pipe.fit(y_train, X_train) # successful
y_pred = pipe_return.predict(fh=fh, X=X_test) # can only be done with X_test during predict, which I won't have
```

In the second, I tried including `WindowSummarizer` directly within `make_reduction`, but I wasn't able to fit to my `X_train` and `y_train`
```python
forecaster = make_reduction(
    regressor,
    transformers=[WindowSummarizer(**kwargs)],
    window_length=None,
    strategy="recursive",
    pooling="global",
)
forecaster.fit(y_train, X_train) # didn't work!
```

I noted this class `RecursiveReductionForecaster` which looks like it might fit my needs but I'm not sure how to combine this with `WindowSummarizer`.

Any helps would be appreciated, thanks in advance!

**Comment by eschwarzbeckf** [comment]
hello, I think you will need `X_test`  to predict, since the model was trained with exogenous values.

You could either:
1. Reuse some of your `X_train` values in future (sort of simulate similar values)
2. Remove the exogenous variables from the model

**_Note_** if your exogenous (`X`) variables are lagged values or other calculations using your endogenous (`y`), the parameter in `transformers=[WindowSummarizer(**kwargs)]` will calculate those values as exogenous and reuse them for future predictions. You can validate this in your second approach:

```
forecaster = make_reduction(
    regressor,
    transformers=[WindowSummarizer(**kwargs)],
    window_length=None,
    strategy="recursive",
    pooling="global",
)
forecaster.fit(y_train) # --> Remove the X_train

# name of the columns used to make predictions
forecaster.estimator_.feature_names_in_

```
